ssz/src/main/java/org/ethereum/beacon/ssz/type/SSZBitListType.java

```diff
@@ -0,0 +1,53 @@
+package org.ethereum.beacon.ssz.type;
+
+import org.ethereum.beacon.ssz.access.SSZField;
+import org.ethereum.beacon.ssz.access.SSZListAccessor;
+import org.ethereum.beacon.ssz.annotation.SSZ;
+
+/**
+ * {@link tech.pegasys.artemis.util.collections.Bitlist} and {@link
+ * tech.pegasys.artemis.util.collections.Bitvector}
+ *
+ * <p>It's like a list/vector, but we have elements inside that are smaller than our usual elements:
+ * bits instead of bytes. So we mimic old interfaces in bytes and add some new interfaces in bits
+ */
+public class SSZBitListType extends SSZListType {
+
+  public SSZBitListType(
+      SSZField descriptor,
+      TypeResolver typeResolver,
+      SSZListAccessor accessor,
+      int vectorLength,
+      long maxSize) {
+    super(descriptor, typeResolver, accessor, vectorLength, maxSize);
+  }
+
+  /**
+   * If this type represents SSZ Vector then this method returns its length.
+   *
+   * @see SSZ#vectorLength()
+   * @see SSZ#vectorLengthVar()
+   */
+  @Override
+  public int getVectorLength() {
+    return (int) fromAtomicSize(super.getVectorLength());
+  }
+
+  public int getBitSize() {
+    return super.getVectorLength();
+  }
+
+  private long fromAtomicSize(long atomicSize) {
+    int addon = getType() == Type.LIST ? 8 : 7;
+    return atomicSize == SSZType.VARIABLE_SIZE ? atomicSize : (atomicSize + addon) / 8;
+  }
+
+  @Override
+  public long getMaxSize() {
+    return fromAtomicSize(super.getMaxSize());
```

Comment:
This call makes a side effect. When lengths is checked against max size it counts bytes required to pack bits of decoding bitlist instead of number of actual bits encoded. It could potentially lead to a false negative result of this check, but taking in account that max size is a power of two we have a very narrow surface on which this side effect takes place, i.e. it could fail only if max size is less than eight bits.

I would not insist but it's highly recommend to keep invariant on `vectorLength` and `maxSize` behaviour. Who knows what other misleading behavior it could lead to.

Reply:
It's not a `maxSize` method of Bitlist, it's `maxSize` of ssz type. And the same as above: raw units from it are not needed by anyone on ssz level